### PR TITLE
fix: make data retrieval of IOLink sensor more robust

### PR DESCRIPTION
**Pull request by bossenti**
### Purpose

This PR makes the data retrieval within the IOLink adapter more robust. Prior to it, an error occurring for one port stopped the processing of all subsequent ports. This means, that in case the processing fails for port 2, e.g., since this port is not connected to a sensor and therefore does not deliver any data, parsing for port 3 and 4 is skipped. 
From now on, only the affected port will be skipped.

### Remarks

PR introduces (a) breaking change(s): no

PR introduces (a) deprecation(s): no

### CODE REVIEW

File: streampipes-extensions/streampipes-connect-adapters-iiot/src/main/java/org/apache/streampipes/connect/iiot/adapters/iolink/IfmAlMqttAdapter.java [hunk 1]
```diff
@@ -110,18 +110,26 @@ public void onAdapterStarted(IAdapterParameterExtractor extractor,
...
-
-                var parsedEvent = sensor.parseEvent(eventData);
-                parsedEvent.put("timestamp", System.currentTimeMillis() + i);
-                parsedEvent.put("port", "port" + ports.get(i));
-                parsedEvent.put(SensorVVB001.IO_LINK_MASTER_SN, serialnumber);
-
-                collector.collect(parsedEvent);
+                try {
+                  var portResult = getMap(payload,
+                          "/iolinkmaster/port[%s]/iolinkdevice/pdin".formatted(ports.get(i)));
```
Comment (dominikriemer) [inline streampipes-extensions/streampipes-connect-adapters-iiot/src/main/java/org/apache/streampipes/connect/iiot/adapters/iolink/IfmAlMqttAdapter.java:115]: Can we check if the port exists here before fetching it?

File: streampipes-extensions/streampipes-connect-adapters-iiot/src/main/java/org/apache/streampipes/connect/iiot/adapters/iolink/IfmAlMqttAdapter.java [hunk 2]
```diff
@@ -110,18 +110,26 @@ public void onAdapterStarted(IAdapterParameterExtractor extractor,
...
+                  var eventData = (String) portResult.get("data");
```
Comment (dominikriemer) [inline streampipes-extensions/streampipes-connect-adapters-iiot/src/main/java/org/apache/streampipes/connect/iiot/adapters/iolink/IfmAlMqttAdapter.java:116]: same here

**Review by dominikriemer — CHANGES_REQUESTED** [review]

File: streampipes-extensions/streampipes-connect-adapters-iiot/src/main/java/org/apache/streampipes/connect/iiot/adapters/iolink/IfmAlMqttAdapter.java (hunk 1 above)
Comment (bossenti) [reply streampipes-extensions/streampipes-connect-adapters-iiot/src/main/java/org/apache/streampipes/connect/iiot/adapters/iolink/IfmAlMqttAdapter.java]: Done

File: streampipes-extensions/streampipes-connect-adapters-iiot/src/main/java/org/apache/streampipes/connect/iiot/adapters/iolink/IfmAlMqttAdapter.java (hunk 2 above)
Comment (bossenti) [reply streampipes-extensions/streampipes-connect-adapters-iiot/src/main/java/org/apache/streampipes/connect/iiot/adapters/iolink/IfmAlMqttAdapter.java]: Done

**Review by dominikriemer — APPROVED** [review]

**Pull request merged**